src/ToolBox/SOS/Strike/util.cpp

```diff
@@ -6363,30 +6369,50 @@ HRESULT SymbolReader::LoadSymbols(IMetaDataImport * pMD, ULONG64 baseAddress, __
     if (Status != S_OK)
         return Status;
 
-    char szName[mdNameLen];
     WideCharToMultiByte(CP_ACP, 0, pModuleName, (int) (_wcslen(pModuleName) + 1),
-            szName, mdNameLen, NULL, NULL);
-    return !loadSymbolsForModuleDelegate(szName);
+            m_szModuleName, mdNameLen, NULL, NULL);
+    return !loadSymbolsForModuleDelegate(m_szModuleName);
 #endif // FEATURE_PAL
 }
 
 HRESULT SymbolReader::GetNamedLocalVariable(ISymUnmanagedScope * pScope, ICorDebugILFrame * pILFrame, mdMethodDef methodToken, ULONG localIndex, __inout_ecount(paramNameLen) WCHAR* paramName, ULONG paramNameLen, ICorDebugValue** ppValue)
 {
     HRESULT Status = S_OK;
+#ifdef FEATURE_PAL
+    if (getLocalVariableNameDelegate == nullptr)
+        Status = LoadCoreCLR();
+    if (Status != S_OK)
+        return Status;
+    // FIXME: we need to find a way to release memory after getting string from managed code.
+    WCHAR *wszParamName = new (std::nothrow) WCHAR[mdNameLen];
```

Comment:
The managed code shouldn't free this memory; it should be freed when the function returns.  The managed code shouldn't save this pointer in any global state.  If it is, then maybe the managed code needs to change.

If you absolutely have to save the string in the managed code, then you could allocate is a bstr with SysAllocString and use Marshal.FreeBSTR to free it managed code.  But I don't remember if this is supported in the xplat platforms.